src/wallet/wallet.cpp

```diff
@@ -672,6 +672,23 @@ void CWallet::AddToSpends(const CWalletTx& wtx, WalletBatch* batch)
         AddToSpends(txin.prevout, wtx.GetHash(), batch);
 }
 
+void CWallet::UpdateAddressBookUsed(const CWalletTx& wtx)
+{
+    for (const auto& output : wtx.tx->vout) {
+        CTxDestination dest;
+        if (!ExtractDestination(output.scriptPubKey, dest)) continue;
+        m_address_book[dest].m_used = true;
```

Comment:
It seems to me this is called whenever new item added to `mapWallet` including spends from the wallet. If so, then this will add all outputs of spending txs to the address book.

Reply:
Yes, that's what it's supposed to do.

Reply:
I don't think we should automatically add to the address book all addresses we're sending to

Reply:
re: https://github.com/bitcoin/bitcoin/pull/22693#discussion_r1021153673

> I don't think we should automatically add to the address book all addresses we're sending to

I think i understand why this is being done after seeing the original PR https://github.com/bitcoin/bitcoin/pull/15987

The goal of the PR is to warn the user if they are sending to an address they have previously sent to. The address book is a convenient place to store this information.

Originally looking at this PR, I thought the goal just to make it easy to look up whether a receiving address had been used, in which case `dest` should already be in `m_address_book` here. But even for receiving addresses it can make sense in some cases to set `m_address_book[dest].m_used = true` even if `dest` is is not in `m_address_book` , for example if this is a backup wallet that is being resynced and the address book is incomplete.

Reply:
This would conflate user entered data with programmatically generated data. Address book today is managed by the user. If we want to store a list of all addresses we sent to we should pick a new storage for that.